tests/Unit/PointwiseFunctions/GeneralRelativity/Test_IndexManipulation.cpp

```diff
@@ -8,49 +8,75 @@
-      used_for_size);
+  using Index0 =
+      Tensor_detail::TensorIndexType<Dim, UpOrLo, Frame::Inertial, Index>;
+  using Index1 =
+      Tensor_detail::TensorIndexType<Dim, UpLo::Lo, Frame::Inertial, Index>;
+  Tensor<DataType, Symmetry<2, 1, 1>,
+         index_list<change_index_up_lo<Index0>, Index1, Index1>> (*f)(
+      const Tensor<DataType, Symmetry<2, 1, 1>,
+                   index_list<Index0, Index1, Index1>>&,
+      const Tensor<DataType, Symmetry<1, 1>,
+                   index_list<change_index_up_lo<Index0>,
+                              change_index_up_lo<Index0>>>&) =
+      &raise_or_lower_first_index<DataType, Index0, Index1>;
+  pypp::check_with_random_values<1>(f, "TestFunctions",
+                                    "raise_or_lower_first_index",
+                                    {{{-10., 10.}}}, used_for_size);
 }
 
 template <size_t Dim, UpLo UpOrLo, IndexType Index, typename DataType>
 void test_raise_or_lower(const DataType& used_for_size) {
-  pypp::check_with_random_values<1>(
-      &raise_or_lower_index<DataType, Tensor_detail::TensorIndexType<
-                                          Dim, UpOrLo, Frame::Inertial, Index>>,
-      "numpy", "matmul", {{{-10., 10.}}}, used_for_size);
+  using Index0 =
+      Tensor_detail::TensorIndexType<Dim, UpOrLo, Frame::Inertial, Index>;
+  Tensor<DataType, Symmetry<1>, index_list<change_index_up_lo<Index0>>> (*f)(
+      const Tensor<DataType, Symmetry<1>, index_list<Index0>>&,
+      const Tensor<DataType, Symmetry<1, 1>,
+                   index_list<change_index_up_lo<Index0>,
+                              change_index_up_lo<Index0>>>&) =
+      &raise_or_lower_index<DataType, Index0>;
+  pypp::check_with_random_values<1>(f, "numpy", "matmul", {{{-10., 10.}}},
+                                    used_for_size);
 }
 template <size_t Dim, UpLo UpLo0, UpLo UpLo1, typename Fr,
           IndexType TypeOfIndex, typename DataType>
 void test_trace_last_indices(const DataType& used_for_size) {
-  pypp::check_with_random_values<1>(
-      &trace_last_indices<
-          DataType, Tensor_detail::TensorIndexType<Dim, UpLo0, Fr, TypeOfIndex>,
-          Tensor_detail::TensorIndexType<Dim, UpLo1, Fr, TypeOfIndex>>,
-      "TestFunctions", "trace_last_indices", {{{-10., 10.}}}, used_for_size);
+  using Index0 = Tensor_detail::TensorIndexType<Dim, UpLo0, Fr, TypeOfIndex>;
+  using Index1 = Tensor_detail::TensorIndexType<Dim, UpLo1, Fr, TypeOfIndex>;
+  Tensor<DataType, Symmetry<1>, index_list<Index0>> (*f)(
+      const Tensor<DataType, Symmetry<2, 1, 1>,
+                   index_list<Index0, Index1, Index1>>&,
+      const Tensor<DataType, Symmetry<1, 1>,
+                   index_list<change_index_up_lo<Index1>,
+                              change_index_up_lo<Index1>>>&) =
+      &trace_last_indices<DataType, Index0, Index1>;
+  pypp::check_with_random_values<1>(f, "TestFunctions", "trace_last_indices",
+                                    {{{-10., 10.}}}, used_for_size);
```

Comment:
Because there are `not_null` overloads the compiler can't tell which one you want from just `&func_name`, you need to tell it the type so it can figure it out. The alternative is a `static_cast<...>(&func_name)` in the `pypp` call but I found that even more difficult to read :)